### PR TITLE
[Issue-55]: Gradlew test fails

### PR DESCRIPTION
**Pull request by athkalia**
Rewrote the robolectric tests to make sure mocks are properly injected and no network calls are performed. Using dagger mock library for that
While I was at it I also:
-Upped the gradle version
-Added Butterknife view actions
-Added constraint layout

### CODE REVIEW

File: app/src/main/java/com/example/features/dashboard/dagger/MainActivityModule.java [hunk 1]
```diff
@@ -12,40 +12,36 @@
...
 @Module
-public final class MainActivityModule {
-
-    private MainActivityModule() {
-        throw new AssertionError();
-    }
+public class MainActivityModule {
 
     @Provides
-    public static MainActivityViewState providesMainActivityViewState() {
```
Comment (miguelslemos) [inline app/src/main/java/com/example/features/dashboard/dagger/MainActivityModule.java:27]: why remove static from method?

Comment (athkalia) [reply app/src/main/java/com/example/features/dashboard/dagger/MainActivityModule.java:27]: I just removed them from that whole module, as now it needs to be instantiated by DaggerMock with a constructor call in MainPresenterTest. 
a) I didn't check if DaggerMock works with static providers
b) I removed them for the whole module even though we only mock the presenter object for the sake of simplicity

Comment (miguelslemos) [reply app/src/main/java/com/example/features/dashboard/dagger/MainActivityModule.java:27]: DaggerMock work fine with static

Comment (athkalia) [reply app/src/main/java/com/example/features/dashboard/dagger/MainActivityModule.java:27]: I just tried this, it daggerMock doesn't work with static.

Comment (miguelslemos) [reply app/src/main/java/com/example/features/dashboard/dagger/MainActivityModule.java:27]: @athkalia Sorry, I had left only this method as non-static

**Pull request merged**